engine/src/main/java/org/camunda/bpm/engine/impl/dmn/transformer/DecisionDefinitionHandler.java

```diff
@@ -36,10 +47,24 @@ protected DmnDecisionImpl createFromDecision(DmnElementTransformContext context,
 
     String category = context.getModelInstance().getDefinitions().getNamespace();
     decisionDefinition.setCategory(category);
-    decisionDefinition.setHistoryTimeToLive(ParseUtil.parseHistoryTimeToLive(decision.getCamundaHistoryTimeToLiveString()));
+    decisionDefinition.setHistoryTimeToLive(getEffectiveHistoryTimeToLive(decision));
     decisionDefinition.setVersionTag(decision.getVersionTag());
 
     return decisionDefinition;
   }
 
-}
+  protected Integer getEffectiveHistoryTimeToLive(Decision decision) {
+
+    if (!hasNullTTLConfigured(decision)) {
+      return parseFunction.apply(decision.getCamundaHistoryTimeToLiveString());
+    }
+
+    return configuredHistoryTTL;
+  }
+
+  private boolean hasNullTTLConfigured(Decision decision) {
+    return decision.getCamundaHistoryTimeToLiveString() == null
+        || "null".equals(decision.getCamundaHistoryTimeToLiveString());
+  }
```

Comment:
❓ As far as I can see, the `ParseUtil.parseHistoryTimeToLive()` method can deal with `null` values. Why is this `null` check necessary? In which cases is the `camundaHistoryTimeToLiveString` equal to `"null"`?

Reply:
Probably never, i used it as a safety-net should this value end up being "null" due to some xml parsing. I will remove it for now.